app/code/Magento/InventoryInStorePickup/Plugin/InventoryApi/SourceRepository/SaveInStorePickupPlugin.php

```diff
@@ -19,17 +19,18 @@ class SaveInStorePickupPlugin
      * @param SourceInterface $source
      *
      * @return array
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
      */
     public function beforeSave(
         SourceRepositoryInterface $subject,
         SourceInterface $source
     ):array {
         $extensionAttributes = $source->getExtensionAttributes();
 
-        if ($extensionAttributes->getInStorePickup() !== null) {
-            $source->setInStorePickup($extensionAttributes->getInStorePickup());
+        if ($extensionAttributes->getIsPickupLocationActive() !== null) {
```

Comment:
According to the `SourceInterface` contract `getExtensionAttributes` may also return null.

Reply:
Added check for `null`.